src/charm.py

```diff
@@ -87,7 +116,19 @@ def _get_arch_cli_tools(self, members, name):
             if tarinfo.name == name:
                 yield tarinfo
 
+    def _get_service_status(self, service_name):
+        try:
+            result = subprocess.run(
+                ["systemctl", "is-active", service_name],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            return result.returncode
+        except subprocess.CalledProcessError:
+            log.exception(f"Failed to get {service_name} status.")
```

Comment:
Something here feels weird.  

* if you don't pass `check=True`, then `CalledProcessError` will never be raised.
* i don't see you doing anything with the piped output (`result.stdout` nor `result.stderr`)

Would this work?
```suggestion
        """checks if service is active, returns 0 on success, otherwise non-zero value."""
        return subprocess.call(["systemctl", "is-active", service_name])
```